h2o-core/src/main/java/hex/Model.java

```diff
@@ -2225,23 +2226,39 @@ public boolean testJavaScoring(Frame data, Frame model_predictions, EasyPredictM
 
   public void deleteCrossValidationModels() {
     if (_output._cross_validation_models != null) {
+      Log.info("Cleaning up CV Models for " + this._key.toString());
+      int c = 0;
       for (Key k : _output._cross_validation_models) {
         Model m = DKV.getGet(k);
-        if (m!=null) m.delete(); //delete all subparts
+        if (m!=null) {
+          m.delete(); //delete all subparts
+          c++;
+        }
       }
+      Log.info(c+" models were removed");
+      _output._cross_validation_models = null;
     }
   }
 
   public void deleteCrossValidationPreds() {
     if (_output._cross_validation_predictions != null) {
+      Log.info("Cleaning up CV Predictions for " + this._key.toString());
+      int c = 0;
       for (Key k : _output._cross_validation_predictions) {
         Frame f = DKV.getGet(k);
-        if (f!=null) f.delete();
+        if (f!=null) {
+          f.delete();
+          c++;
+        }
       }
+      Log.info(c+" predictions were removed");
     }
+    _output._cross_validation_predictions = null;
+
     if (_output._cross_validation_holdout_predictions_frame_id != null) {
       _output._cross_validation_holdout_predictions_frame_id.remove();
     }
+    _output._cross_validation_holdout_predictions_frame_id = null;
```

Comment:
Same comments as above.

Reply:
closed

Reply:
reintroducing the null here, but with the update this time: this is used only by tests and AutoML currently, and nullifying only in AutoML through a parameter like `deleteCrossValidationPreds(boolean nullify)` multiplies the amount of almost unused methods.